crates/subspace-node/src/command.rs

```diff
@@ -87,6 +88,23 @@ impl SubstrateCli for Cli {
     }
 }
 
+fn set_default_ss58_version<C: AsRef<dyn ChainSpec>>(chain_spec: C) {
+    chain_spec
+        .as_ref()
+        .properties()
+        .get("ss58Format")
+        .map(|v| {
+            v.as_u64()
+                .expect("ss58Format must always be an unsigned number; qed")
+        })
+        .map(|v| {
+            v.try_into()
+                .expect("ss58Format must always be within u16 range; qed")
+        })
+        .map(Ss58AddressFormat::custom)
```

Comment:
This should never happen and is wrong and should be detected early if bad thing happens.

Reply:
What exactly, missing `ss58Format`? It is only present in `testnet` chainspec.

Reply:
Not missing, but what if I somehow use a random unrelated prefix in `SS58Format` by accident. `Ss58AddressFormat::custom` can use any arbitrary prefix, but only the subspace prefixes make sense in our case in fact, so using another unrelated prefix should be forbidden. Using the substrate account fallback makes sense to me for the local dev network. This function is generic and should be able to deal with all kinds of network chainspec.

Reply:
Well, we can use `TryInto` and turn it into `Ss58AddressFormatRegistry`, but we'll have to turn it back into `Ss58FormatAddress` after that, so this didn't make much sense to me. Also there is nothing inherently wrong with using custom prefix if it was specified in the chainspec, it will be consistent with what user will see in Polkadot.js apps, so it makes sense to just leave it as is and not worry about it at all.

Reply:
Being consistent with polkadot.js apps does mean it's right. If it's wrong, it shouldn't be started. My version with a cost of extra trait prevents this edge case in imagination while yours opens a window in theory. But again, what we are discussing here is really unlikely to happen, so either way works for me, I don't care about it and just approved.